lib/auth/auth_with_roles.go

```diff
@@ -4833,6 +4833,27 @@ func (a *ServerWithRoles) UpsertLock(ctx context.Context, lock types.Lock) error
 	if err := a.action(apidefaults.Namespace, types.KindLock, types.VerbCreate, types.VerbUpdate); err != nil {
 		return trace.Wrap(err)
 	}
+
+	if lock.CreatedBy() == "" {
+		var createdBy string
+		hasAdmin := a.hasBuiltinRole("Admin")
+		if hasAdmin {
+			createdBy = "admin"
+		} else {
```

Comment:
Try removing the else here. Something like:

```
createdBy := string(types.RoleAdmin)
if !hasAdmin {
    createdBy = ...
}
```

Reply:
Good call, done